app/src/main/java/org/onionshare/android/files/FileManager.kt

```diff
@@ -195,7 +194,16 @@ class FileManager @Inject constructor(
         val currentState = state.value
         if (currentState is Zipping) currentState.zip.delete()
         if (currentState is Zipped) currentState.sendPage.zipFile.delete()
-        _state.value = Added(currentState.files)
+        _state.value = if (currentState is FilesState.Error) {
+            FilesState.Error(currentState.files, currentState.errorFile)
+        } else {
+            Added(currentState.files)
```

Comment:
When stopping sharing after an error, this would clean our error state too quickly for showing something in the UI.